fix(orders-table): display allowance warning in Safe

shoom3301 opened the pull request:
# Summary

Fixes #4076

`hasValidPendingPermit` can be `undefined`

<img width="1027" alt="image" src="https://github.com/user-attachments/assets/177bb5a0-547f-417f-8a94-68803e8a63b6" />

# To Test

This issue was introduced in #3487, so please make sure this issue is not back.

See #4076

elena-zh commented:
Hey @shoom3301 , everything looks good besides one bug (#3480) that has become reproducible here again

shoom3301 commented:
@elena-zh thank you! Fixed that.
@alfetopito could you check [the fix](https://github.com/cowprotocol/cowswap/pull/5207/commits/0c75924dad99375e41a838cd63dd91d72325da81) please?
Mainly, I got rid of `PendingPermitUpdater` and everything related and replaced the logic with `getOrderPermitAmount()`.
Now `_hasEnoughBalanceAndAllowance` takes an order permit value into account.

shoom3301 commented on apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrderRow/index.tsx:
```diff
@@ -173,10 +171,10 @@ export function OrderRow({
   )
 
-  const withAllowanceWarning = hasEnoughAllowance === false && hasValidPendingPermit === false
+  const withAllowanceWarning = hasEnoughAllowance === false
```
Now it's simplier, `hasEnoughAllowance` already includes result of permit checks

shoom3301 commented on apps/cowswap-frontend/src/modules/ordersTable/utils/getOrderParams.ts:
```diff
@@ -43,7 +45,7 @@ export function getOrderParams(
     sellAmount,
     balance,
-    allowance,
+    allowance: getBiggerAmount(allowance, permitAmount),
```
I take allowance value from blockchain OR permit value from order appData, depending on what is bigger

elena-zh reviewed (approved):
Works good now

anxolin commented on apps/cowswap-frontend/src/utils/orderUtils/getOrderPermitAmount.ts:
```diff
@@ -0,0 +1,34 @@
+          decoded.spender?.toLowerCase() === spenderAddress &&
+          decoded.owner?.toLowerCase() === order.owner.toLowerCase() &&
+          (decoded.deadline as BigNumber)?.toNumber() > Date.now() / 1000
+        )
```
I still don't get it. I think the try catch is outside of the map function. If you throw inside a map, it propagates the error up.

Did a small test, just in case:

<img width="573" alt="image" src="https://github.com/user-attachments/assets/d53b3325-d852-42c4-935d-b549bdc55794" />

Maybe I'm missing something, but shouldn't you have the try-catch inside the map function? Probably, if you want to do it like this, could be a reduce function which only adds it to the result if it succeeds with the decoding

Also, probably all these callsdata start with the same prefix, the name of the encoded name of the function permit. 
Wouldn't it be good to do some prefiltering so we don't try to decode other hooks?

shoom3301 replied:
You are right, I'm sorry!
Fixed it
